src/hotspot/share/runtime/java.cpp

```diff
@@ -364,7 +357,9 @@ void print_statistics() {
 
   ThreadsSMRSupport::log_statistics();
 
-  log_vm_stats(tty);
+  if (log_is_enabled(Info, perf, class, link)) {
+      ClassLoader::print_counters(tty);
+  }
```

Comment:
Again this needs to comment why we check the log is active but write to tty instead of the logstream.

Or we could put the `log_is_enabled` check inside `print_counters` rather than require callers to do it. That also allows us to print that the counters are disabled eg.
```
void ClassLoader::print_counters(outputStream *st) {
  // The counters are only active if the logging is enabled, but
  // we print to the passed in outputStream as requested.
  if (log_is_enabled(Info, perf, class, link)) {
    st->print_cr("ClassLoader:");
    ...
  } else {
    st->print_cr("ClassLoader: <not enabled>");
  }
```

Reply:
I've made the above changes as you suggested without the else part to be consistent with printing of other statistics.